include/zephyr/net/ieee802154_radio.h

```diff
@@ -757,61 +757,171 @@ enum ieee802154_config_type {
-	 *     ----------------------------------------------------------[ CSL window ]-----+
-	 *                                             ^                                    |
-	 *                                             |                                    |
-	 *                                             +--------------------- loop ---------+
+	 * Enables or disables a device as a CSL receiver and configures its CSL
+	 * period.
+	 *
+	 * @details Configures the CSL period in units of 10 symbol periods.
+	 * Values greater than zero enable CSL if the driver supports it and the
+	 * device starts to operate as a CSL receiver. Setting this to zero
+	 * disables CSL on the device. If the driver does not support CSL, the
+	 * configuration call SHALL return -ENOTSUP.
+	 *
+	 * See section 7.4.2.3 and section 8.4.3.6, table 8-104, macCslPeriod.
+	 *
+	 * @note Confusingly the standard calls the CSL receiver "CSL
+	 * coordinator" (i.e. "coordinating the CSL protocol timing", see
+	 * section 6.12.2.2), although, typically, a CSL coordinator is NOT also
+	 * an IEEE 802.15.4 FFD coordinator or PAN coordintor but a simple RFD
+	 * end device (compare the device roles outlined in sections 5.1, 5.3,
+	 * 5.5 and 6.1). To avoid confusion we therefore prefer calling CSL
+	 * coordinators (typically an RFD end device) "CSL receivers" and CSL
+	 * peer devices (typically FFD coordinators or PAN coordinators) "CSL
+	 * transmitters". Also note that at this time, we do NOT support
+	 * unsynchronized transmission with CSL wake up frames as specified in
+	 * section 6.12.2.4.4.
+	 *
+	 * To offload CSL receiver timing to the driver the upper layer SHALL
+	 * combine several configuration options in the following way:
+	 *
+	 * 1. Use @ref IEEE802154_CONFIG_ENH_ACK_HEADER_IE once with an
+	 *    appropriate pre-filled CSL IE and the CSL phase set to zero.  The
+	 *    CSL phase SHALL be injected on-the-fly by the driver at runtime as
+	 *    outlined in 2. below. Adding a short and extended address will
+	 *    inform the driver of the specific CSL receiver to which it SHALL
+	 *    inject CSL IEs. If no addresses are given then the CSL IE will be
+	 *    injected into all enhanced ACK frames as soon as CSL is enabled.
+	 *    This configuration SHALL be done before enabling CSL by setting a
+	 *    CSL period greater than zero.
+	 *
+	 * 2. Configure @ref IEEE802154_CONFIG_EXPECTED_RX_TIME immediately
+	 *    followed by @ref IEEE802154_CONFIG_CSL_PERIOD. To prevent race
+	 *    conditions, the upper layer SHALL ensure that the receiver is not
+	 *    enabled during or between the two calls (e.g. by a previously
+	 *    configured RX slot) nor SHALL a frame be transmitted concurrently.
+	 *
+	 *    The expected RX time SHALL point to the end of SFD of an ideally
+	 *    timed RX frame in the nearest CSL channel sample (i.e.  whose "end
+	 *    of SFD" lies exactly at the center of the CSL channel sample).
```

Comment:
This is not the case of the current OpenThread implementation. See these configuration options: https://github.com/zephyrproject-rtos/openthread/blob/d62167ee34b091e7025c9ec2820aae71e17a3944/src/core/config/mac.h#L491-L513. Basically the channel sample is asymmetrical around `IEEE802154_CONFIG_EXPECTED_RX_TIME`.

Reply:
Fixed.